src/Adapter/Order/CommandHandler/AddCartRuleToOrderHandler.php

```diff
@@ -80,6 +81,7 @@ public function handle(AddCartRuleToOrderCommand $command): void
         $this->contextStateManager
             ->setCurrency(new Currency($order->id_currency))
             ->setCustomer(new Customer($order->id_customer))
+            ->setTranslatorLocale((new Language($order->id_lang))->locale)
```

Comment:
```suggestion
            ->setLanguage((new Language($order->id_lang)))
```

Reply:
Once `setLanguage` handles the translator's locale change it becomes easier to use here